### ISSUE TITLE
Store initial state for ledger channels

### ISSUE THREAD

**Issue by lalexgap** [statechannels/statechannels#3061]
We need to store the initial state for the implementation of `challenging-v0` (see https://www.notion.so/Direct-funding-f0fe906bb6fd4a298602b140c3bb0899#8b1745c0097b4957a0ae435d2e21f6bd)

We also need to think about migration for the channels created before we started storing this.

**Comment by andrewgordstewart** [comment]
For discussion: We may consider storing the `s0` state in the channel manager. If we did this, then it probably implies:
- the wallet should expose the chain service (so we can call `challenge`)

This implication smells bad. But, the alternative might be we have to change channel storage in the wallet to implement `challenging-v0`, which also smells bad.

**Comment by lalexgap** [comment]
> For discussion: We may consider storing the `s0` state in the channel manager. If we did this, then it probably implies:
> 
> * the wallet should expose the chain service (so we can call `challenge`)
> 
> This implication smells bad. But, the alternative might be we have to change channel storage in the wallet to implement `challenging-v0`, which also smells bad.

Based on our discussion the `ChannelManager` will be responsible for storing `s0`. 

Closing this in favour of https://github.com/statechannels/the-graph/issues/487
